cluster-api-gcp/deploy/deploy_helper.go

```diff
@@ -286,18 +292,21 @@ func (d *deployer) waitForApiserver(master string, timeout time.Duration) error
 	}
 	client := &http.Client{Transport: tr}
 
-	startTime := time.Now()
-
 	var err error
 	var resp *http.Response
-	for time.Now().Sub(startTime) < timeout {
+	eb := util.NewExponentialBackoff(timeout, 10 * time.Second)
+	backoff.Retry(func() error {
 		resp, err = client.Get(endpoint)
 		if err == nil && resp.StatusCode == 200 {
 			return nil
 		}
-		time.Sleep(1 * time.Second)
+		return err
+	}, eb)
+	if err != nil {
+		glog.Infof("Error waiting for apiserver: %s", err)
```

Comment:
may be glog.Errorf() ? or better don't log at all and let the caller handle the error message.

Reply:
From what I saw, the deployer script logs the error at the very end, so it's hard to know what exactly failed. I changed the log to Errorf